src/Nordea/Components/Common.elm

```diff
@@ -54,109 +34,26 @@ type alias CharCounter =
     }
 
 
-bottomInfo : InputProperties -> List (Html msg)
-bottomInfo config =
-    let
-        charCounterView =
-            config.charCounter
-                |> Html.viewMaybe
-                    (\charCounter ->
-                        Text.textSmallLight
-                            |> Text.view
-                                [ css [ marginLeft auto, color Colors.grayDark ] ]
-                                [ String.fromInt charCounter.current ++ "/" ++ String.fromInt charCounter.max |> text ]
-                    )
-    in
-    [ div
-        [ css
-            [ displayFlex
-            , flexBasis (pct 100)
-            , justifyContent spaceBetween
-            , marginTop (rem 0.5)
-            , Css.property "gap" "1rem"
-            ]
-        ]
-        [ config.errorMessage
-            |> Html.viewMaybe
-                (\errText ->
-                    Text.textSmallLight
-                        |> Text.view
-                            [ css
-                                [ displayFlex
-                                , alignItems center
-                                , color Colors.redDark
-                                ]
-                            ]
-                            [ Icons.error [ css [ marginRight (rem 0.5), flex none ] ]
-                            , text errText
-                            ]
-                )
-        , charCounterView
-        ]
-        |> showIf (Maybe.isJust config.errorMessage)
-    , div
-        [ css
-            [ displayFlex
-            , flexBasis (pct 100)
-            , justifyContent spaceBetween
-            , marginTop (rem 0.2)
-            , Css.property "gap" "1rem"
-            ]
-        ]
-        [ config.hintText
-            |> Html.viewMaybe
-                (\hintText ->
-                    Text.textSmallLight
-                        |> Text.view [ css [ color Colors.grayDark ] ] [ text hintText ]
-                )
-        , charCounterView |> showIf (Maybe.isNothing config.errorMessage)
-        ]
-        |> showIf (Maybe.isJust config.hintText || Maybe.isJust config.charCounter)
-    ]
+type alias LabelProperties =
+    { label : String
+    , isError : Bool
+    }
 
 
-topInfo : InputProperties -> Html msg
-topInfo config =
-    let
-        toI18NString requirednessHint =
-            case requirednessHint of
-                Mandatory translate ->
-                    translate strings.mandatory
+viewLabel : LabelProperties -> List (Attribute msg) -> Html msg
+viewLabel { label, isError } attrs =
+    Text.textSmallLight
+        |> Text.view
+            ([ css [ color Colors.redDark |> styleIf isError ] ] ++ attrs)
+            [ Html.label [] [ Html.text label ] ]
```

Comment:
This label is not semantically connected to any form controls. It should either have a `for` prop referencing the id of the element, or be wrapped around it. For reusable components, I prefer the latter as you avoid having to enforce that the id is unique across the page. 

I would also suggest to have the Text inside the label rather that having the label inside the text. When using the viewLabel function, one should expect the attributes to be set on the html label and not the span.

Reply:
I'll add the `for` property, good catch🙏  To enforce the uniqueness of it we could prefix with type of field and postfix with random id like Entur has done.
Was a bit uncertain on where the label should be. Sound like a good idea to put it outside the `Text`.

Reply:
Random is cumbersome in Elm tho, all inputs suddenly require an update roundtrip and state 😛

Reply:
😬 I'll try to figure out something that works 🙏